lib/rules/await-async-event.ts

```diff
@@ -0,0 +1,163 @@
+import { ASTUtils, TSESTree } from '@typescript-eslint/utils';
+
+import { createTestingLibraryRule } from '../create-testing-library-rule';
+import {
+	findClosestCallExpressionNode,
+	getFunctionName,
+	getInnermostReturningFunction,
+	getVariableReferences,
+	isPromiseHandled,
+} from '../node-utils';
+import { EVENTS_SIMULATORS } from '../utils';
+
+export const RULE_NAME = 'await-async-event';
+export type MessageIds = 'awaitAsyncEvent' | 'awaitAsyncEventWrapper';
+const FIRE_EVENT_NAME = 'fireEvent';
+const USER_EVENT_NAME = 'userEvent';
+type EventModules = typeof EVENTS_SIMULATORS[number];
+export type Options = [
+	{
+		eventModule: EventModules | EventModules[];
+	}
+];
+
+export default createTestingLibraryRule<Options, MessageIds>({
+	name: RULE_NAME,
+	meta: {
+		type: 'problem',
+		docs: {
+			description: 'Enforce promises from async event methods are handled',
+			recommendedConfig: {
+				dom: ['error', { eventModule: 'userEvent' }],
+				angular: ['error', { eventModule: 'userEvent' }],
+				react: ['error', { eventModule: 'userEvent' }],
+				vue: ['error', { eventModule: 'fireEvent' }],
+				marko: ['error', { eventModule: 'fireEvent' }],
```

Comment:
Vue and Marko actually need both by default: we are aiming for userEvent v14 by default, and fireEvent is async in those frameworks:

```suggestion
				vue: ['error', { eventModule: ['fireEvent', 'userEvent'] }],
				marko: ['error', { eventModule: ['fireEvent', 'userEvent'] }],
```